python/datasources/acs_condition.py

```diff
@@ -0,0 +1,509 @@
+        df_without_condition = df_without_condition.rename(columns={AMOUNT: without_condition_raw_count})
+
+        raw_count = generate_column_name(measure, HAS_ACS_ITEM_SUFFIX)
+        df_with_condition = df_with_condition.rename(columns={AMOUNT: raw_count})
+
+        merge_cols = group_cols.copy()
+        merge_cols.append(std_col.STATE_FIPS_COL)
+        merge_cols.remove(tmp_amount_key)
+        if geo == COUNTY_LEVEL:
+            merge_cols.append(std_col.COUNTY_FIPS_COL)
+
+        df_with_condition = df_with_condition[merge_cols + [raw_count]]
+        df_without_condition = df_without_condition[merge_cols + [without_condition_raw_count]]
+
+        # Generate the population for each condition by adding together
+        # the raw counts of people with and without the condition.
+        population_df = pd.merge(df_without_condition, df_with_condition, on=merge_cols, how='left')
+        population = generate_column_name(measure, POP_SUFFIX)
+        population_df[[raw_count, without_condition_raw_count]] = \
+            population_df[[raw_count, without_condition_raw_count]].astype(float)
+        population_df[population] = population_df[raw_count] + population_df[without_condition_raw_count]
+        population_df = population_df[merge_cols + [population]]
+
+        # Merge the population df back into the df of people with the condition
+        # to create our main df.
+        df = pd.merge(df_with_condition, population_df, on=merge_cols, how='left')
+        df = df[merge_cols + [population, raw_count]]
+        df = update_col_types(df)
+
+        if geo == NATIONAL_LEVEL:
+            groupby_cols = [std_col.AGE_COL]
+            if demo != RACE:
+                groupby_cols.append(std_col.SEX_COL)
+
+            df = df.groupby(groupby_cols).sum().reset_index()
+            df[std_col.STATE_FIPS_COL] = US_FIPS
+
+        groupby_cols = [std_col.STATE_FIPS_COL]
+        if geo == COUNTY_LEVEL:
+            groupby_cols.append(std_col.COUNTY_FIPS_COL)
+
+        if demo == AGE:
+            groupby_cols.append(std_col.AGE_COL)
+        elif demo == SEX:
+            groupby_cols.append(std_col.SEX_COL)
+
+        df = df.groupby(groupby_cols).sum().reset_index()
+
+        # Rename age column and combine needed age ranges
+        if demo == AGE:
+            df[std_col.AGE_COL] = df[std_col.AGE_COL].apply(rename_age_bracket)
+
+            if measure == POVERTY_MEASURE:
+                df[std_col.AGE_COL] = df[std_col.AGE_COL].apply(get_poverty_age_range)
+                df = df.groupby(groupby_cols).sum().reset_index()
+
+        return df
+
+    def post_process(self, df, demo, geo):
+        """Merge population data, state, and county names.
+           Do all needed calculations to generate per100k and pct share
+           columns.
+           Returns a dataframe ready for the frontend.
+
+           df: Dataframe with raw health insurance numbers.
```

Comment:
generalize this doc string to reflect multiple ACS raw counts not just health insurance